### PR TITLE
Javalab: show appropriate options in comment options menu

### PR DESCRIPTION
**Pull request by bencodeorg**
The "comment options" menu is an ellipsis that appear on each comment to show the option to delete a comment (only for teachers) and to resolve/re-open a comment (only for project owner). This PR:

- adds a new redux property `isPeerReviewing` that is passed from the back end when one student is viewing another student's code.
- hides the ellipsis entirely if a student is in this mode.
- if the ellipsis is being shown (ie, not isPeerReviewing), shows the delete option in the comment options menu if the user is a teacher
- if the ellipsis is being shown (ie, not isPeerReviewing) shows the resolve/re-open option (one or the other, depending on the state of the comment) if the user is a student

**Student reviewing another student's work**

image

**Student looking at their own project**

image

**Teacher looking at student work**

image

## Testing story

Tested manually with two students in the same section and a teacher. Also added unit tests.

### CODE REVIEW

File: apps/src/templates/instructions/codeReview/Comment.jsx [hunk 1]
```diff
@@ -45,7 +46,7 @@ export default class Comment extends Component {
   };
 
   renderFormattedTimestamp = timestampString =>
-    moment.utc(timestampString).format('M/D/YYYY [at] h:mm A');
+    moment(timestampString).format('M/D/YYYY [at] h:mm A');
```
Comment (bencodeorg) [inline apps/src/templates/instructions/codeReview/Comment.jsx:49]: Unrelated fix to show timestamps in local time instead of UTC.

File: apps/src/javalab/Javalab.js [hunk 2]
```diff
@@ -136,6 +136,7 @@ Javalab.prototype.init = function(config) {
     channelId: config.channel,
     isProjectLevel: !!config.level.isProjectLevel,
     isEditingStartSources: this.isStartMode,
+    isCodeReviewing: !!config.isCodeReviewing,
```
Comment (sanchitmalhotra126) [inline apps/src/javalab/Javalab.js:139]: Curious how this gets set - will `config.isCodeReviewing` be updated every time a user switches to another student's project via the review dropdown?

File: apps/src/templates/instructions/ReviewTab.jsx [hunk 3]
```diff
@@ -8,7 +10,9 @@ import * as codeReviewDataApi from './codeReview/codeReviewDataApi';
 
 const FLASH_ERROR_TIME_MS = 5000;
 
-export default class ReviewTab extends Component {
+class ReviewTab extends Component {
+  static propTypes = {viewAsCodeReviewer: PropTypes.bool.isRequired};
```
Comment (sanchitmalhotra126) [inline apps/src/templates/instructions/ReviewTab.jsx:15]: nit: add `// populated by redux`?

**Review by sanchitmalhotra126 — APPROVED** [review]
LGTM!

File: apps/src/javalab/Javalab.js (hunk 2 above)
Comment (bencodeorg) [reply apps/src/javalab/Javalab.js:139]: It gets passed in from the controller here and is set as a page constant (I believe such that it cannot be changed once the page is loaded):

https://github.com/code-dot-org/code-dot-org/blob/35c963a410423dfce509189716b325c37a02193e/dashboard/app/controllers/script_levels_controller.rb#L407

As far as I know, we'll be fully reloading the page for one student to view another student's work, so we'd be setting `isCodeReviewing` to true on page load after a student selects another student from the dropdown.
